rllib/algorithms/impala/impala.py

```diff
@@ -865,7 +873,7 @@ def process_experiences_tree_aggregation(
 
     def update_workers_if_necessary(self) -> None:
         # Only need to update workers if there are remote workers.
-        global_vars = {"timestep": self._counters[NUM_AGENT_STEPS_SAMPLED]}
+        global_vars = {"timestep": self._counters[NUM_AGENT_STEPS_TRAINED]}
```

Comment:
Just out of curiosity: These should be the same, unless I use the mixin buffer with ratio > 0.0, correct?
I feel like we don't really have any opinionated rule of when to use sampled steps and when to use trained steps. IMO, we should always count both.

Reply:
Hmmmmm im not sure if this is the case -- We definitely can sample more timesteps than we have trained on at any given moment. 

In practice it doesn't make sense to adjust learning rates and entropy coefficients until the agent has been updated, hence why I changed it here from sampled to trained.